src/api_server/src/lib.rs

```diff
@@ -41,6 +41,7 @@ pub struct ApiServer {
 }
 
 impl ApiServer {
+    #[cfg_attr(feature = "tracing", log_instrument::instrument)]
```

Comment:
I've added details on this to the commit message on `feat: Add code instrumentation`, more details are in https://github.com/JonathanWoollett-Light/clippy-tracing.

Reply:
Right, so this answers the question of how the instrumentation was introduced.
When someone adds a new function, what do they need to do for it to be instrumented? Also what is a mechanism, so they don't forget to do that?

Reply:
To add new instrumentation they can re-run the same command, it can be part of a githook.

The mechanism to ensure it is not forgotten is in [`test: Add test for instrumentation`](https://github.com/firecracker-microvm/firecracker/pull/4063/commits/57d05d1e6a9896547efa21ff6b774d0ee7ed320b) which adds a test that will fail if instrumentation is missing.